backend/api/views/main.py

```diff
@@ -33,21 +33,31 @@ def add_<yourclassname>():
     return create_response(status=200, message="success")
 """
 
-# function that is called when you visit /documetns
+# Gets all documents that can be specified using a query string
 @main.route("/document", methods=["GET"])
 def get_document():
+    # gets database values from query string, if missing is None
     kwargs = {}
     kwargs["fileID"] = request.args.get("fid")
     kwargs["userID"] = request.args.get("uid")
     kwargs["status"] = request.args.get("status")
     kwargs["docClass"] = request.args.get("docClass")
     kwargs["fileName"] = request.args.get("fileName")
     kwargs["latest"] = request.args.get("latest")
+
+    # stores date and time for parse searching
     date = request.args.get("date")
     description = request.args.get("description")
+
+    # refines kwargs to remove all the None values and only search by values provided
     kwargs = {k: v for k, v in kwargs.items() if v is not None}
+
+    # if no search parameters provided return all documents
     if len(kwargs) == 0:
         docs = Document.query.all()
+
+        # Since description and date were not part of search parameters,
+        # refines searches by date and description
         if description is not None:
             docs = [
                 i
```

Comment:
Can you elaborate on what you're doing with description - why it needs to be done separately from the other parameters